app/utils/misc.tsx

```diff
@@ -2,15 +2,40 @@ import { clsx, type ClassValue } from 'clsx'
 import { useEffect, useMemo, useRef, useState } from 'react'
 import { useFormAction, useNavigation } from 'react-router'
 import { useSpinDelay } from 'spin-delay'
+import { GetSrcArgs, defaultGetSrc } from 'openimg/react'
 import { extendTailwindMerge } from 'tailwind-merge'
 import { extendedTheme } from './extended-theme.ts'
 
 export function getUserImgSrc(imageId?: string | null) {
-	return imageId ? `/resources/user-images/${imageId}` : '/img/user.png'
+	return imageId ? `/resources/images?userImageId=${imageId}` : '/img/user.png'
 }
 
 export function getNoteImgSrc(imageId: string) {
-	return `/resources/note-images/${imageId}`
+	return `/resources/images?noteImageId=${imageId}`
+}
+
+export function getImgSrc({
+	height,
+	optimizerEndpoint,
+	src,
+	width,
+	fit,
+	format,
+}: GetSrcArgs) {
+	if (src.startsWith(optimizerEndpoint)) {
+		const [endpoint, query] = src.split('?')
+		const searchParams = new URLSearchParams(query)
+		searchParams.set('h', height.toString())
+		searchParams.set('w', width.toString())
+		if (fit) {
+			searchParams.set('fit', fit)
+		}
+		if (format) {
+			searchParams.set('format', format)
+		}
+		return `${endpoint}?${searchParams.toString()}`
+	}
+	return defaultGetSrc({ height, optimizerEndpoint, src, width, fit, format })
 }
```

Comment:
Any reason this isn't just the default behavior?

Reply:
I would like that adding openimg is just as easy as replacing `img` with `Img`: `<img src="/cat.png" />`  to `<Img src="/cat" />`. That's what I built my logic around. But here, we want our source to specify the optimizer endpoint directly.

Reply:
That makes sense. I think I will add a comment about that.